sdk/core/core-auth/src/tokenCredential.ts

```diff
@@ -52,6 +52,16 @@ export interface GetTokenOptions {
      */
     tracingContext?: Context;
   };
+
+  /**
+   * Allows specifying a tenantId. Useful to handle challenges that provide tenant Id hints.
+   */
+  tenantId?: string;
+
+  /**
+   * If set to true, allows authentication flows to change the tenantId of the request if a different tenantId is received from a challenge or through a direct getToken call.
+   */
+  allowMultiTenantAuthentication?: boolean;
```

Comment:
This shouldn't go on GetTokenOptions but rather should be at to TokenCredentialOptions as it's something users configure when they create the credential.